azure/Guidelines.md

```diff
@@ -0,0 +1,259 @@
+
+The format of the root of the direct endpoint **MUST** be as follows:
+
+```
+http(s)://<tenant-id>-<service-defined-root>.<service>.azure.net
+```
+
+1. A request is made to the default end point (GET or HEAD).  For example:
+
+   ```
+   GET https://blobstore.azure.net/contoso.com/account1/container1/blob2
+   ```
+
+2. That request is returned with the `Content-Location` header set to the direct endpoint.  See [RFC2557]:
+
+   ```
+   200 OK
+   Content-Location: http://contoso-dot-com-account1.blobstore.azure.net/container1/blob2
+   ```
+
+   Or, with the GUID format:
+
+   ```
+   200 OK
+   Content-Location: http://00000000-0000-0000-C000-000000000046-account1.blobstore.azure.net/container1/blob2
+   ```
+
+## Versioning
+
+All Azure APIs **MUST** support explicit versioning.  It's critical that clients can count on services to be stable over time, and it's critical that Azure services can add features and make changes.
+
+The OneAPI guidelines offer a couple of different options on how to specify an API version and guidance on what constitutes a breaking change.  This section of the Azure API guidelines describes which options are required of Azure services as well as some guidance about deprecation policy.  There is also a section about additional versioning practices necessary to support Azure Stack and Azure compatibility.
+
+### Specifying the version in Azure
+
+The OneAPI guidelines give two options for how services and clients communicate the version: a url segment and a query parameter. Azure services **MUST** use the api-version query parameter. For example:
+
+```
+GET http://blobstore.azure.com/foo.com/acct1/c1/blob2?api-version=1.0
+PUT http://blobstore.azure.com/foo.com/acct1/c1/b2?api-version=2014-12-07
+POST http://blobstore.azure.com/foo.com/acct1/c1/b2?api-version=2015-12-07
+```
+
+### Breaking changes in Azure
+
+A breaking change is any change in the API that may cause client or service code making the API call to fail. Obvious examples of such a change are the removal of an endpoint, adding or removing a required field or changing the format of the body (from XML to JSON for example).
+
+Even though we recommend clients ignore new fields, there are many libraries and clients that are strict. Therefore, Azure services **MUST** update the version number of their API even when adding optional fields. In fact, servers should be as strict as possible. Ignoring a field can result in the API accepting content that containered a typo or an element at the wrong level of nesting. If this missing field changes the semantics (for example, we have seen cases where security settings were misplaced and ignored, leaving the resources more exposed than intended) this can be a huge and hard to discover error.
```

Comment:
strict isn't the correct word here.   There are many clients that do not abide by this recommendation might be a better phrase.  'strict' im,plies they are doing something more correct which is hardly the case,

Reply:
I;ve reworded, but I don't like the new wording - this is caused by the mash-up of the updated breaking change guidance into the new doc.  We should revisit this during the reconciliation and update meetings.